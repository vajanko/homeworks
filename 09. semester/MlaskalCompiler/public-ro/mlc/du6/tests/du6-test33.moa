Integer constants
0:	1
1:	2
2:	3
3:	4
4:	5
5:	6

Real constants

String constants

Identifiers
0:	"BOOLEAN"
1:	"INTEGER"
2:	"REAL"
3:	"STRING"
4:	"FALSE"
5:	"TRUE"
6:	"READI"
7:	"READR"
8:	"READS"
9:	"P"
10:	"WRITEI"
11:	"WRITER"
12:	"WRITES"
13:	"WRITELN"
14:	"DU6TEST"

Functions and procedures
READI:INTEGER	[0;2], used registers=0
READR:REAL	[3;5], used registers=0
READS:STRING	[6;8], used registers=0
WRITEI(P:INTEGER [size=1;stkoffs=-1])	[9;11], used registers=0
WRITER(P:REAL [size=1;stkoffs=-1])	[12;14], used registers=0
WRITES(P:STRING [size=1;stkoffs=-1])	[15;17], used registers=0
WRITELN	[18;19], used registers=0
DU6TEST	[20;21], used registers=0

Instructions
READI:
0:	FNCREADI
1:	LSTI	[-1]
2:	RET
READR:
3:	FNCREADR
4:	LSTR	[-1]
5:	RET
READS:
6:	FNCREADS
7:	LSTS	[-1]
8:	RET
WRITEI:
9:	LLDI	[-1]
10:	FNCWRITEI
11:	RET
WRITER:
12:	LLDR	[-1]
13:	FNCWRITER
14:	RET
WRITES:
15:	LLDS	[-1]
16:	FNCWRITES
17:	RET
WRITELN:
18:	FNCWRITELN
19:	RET
DU6TEST:
20:	CALL	18	;; label=WRITELN
21:	HALT

