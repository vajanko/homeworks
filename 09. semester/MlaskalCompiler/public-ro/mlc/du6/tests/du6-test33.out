Warning (line 3): Label '1' defined but never used
Warning (line 3): Label '2' declared but never used
Warning (line 3): Label '3' declared but never used
Warning (line 3): Label '4' declared but never used
Warning (line 3): Label '5' declared but never used
Warning (line 3): Label '6' declared but never used
Everything is nearly OK, although there were 6 warnings
